app/services/link_reporter_csv_service.rb

```diff
@@ -0,0 +1,70 @@
+class LinkReporterCsvService
+  delegate :<<, :close, to: :csv
+
+  def initialize(reports_dir:, organisation:)
+    @organisation = organisation
+    @reports_dir = reports_dir
+  end
+
+  def generate
+    csv = CSV.open(file_path, "w", encoding: "UTF-8")
+    csv << headings
+    public_editions.each { |edition| add_edition_row(csv, edition) }
+  ensure
+    csv.close
+  end
+
+private
+
+  attr_reader :organisation, :reports_dir
+
+  def public_editions
+    Edition.publicly_visible.with_translations.in_organisation(organisation)
+  end
+
+  def file_path
+    reports_dir.join("#{@organisation.slug}_links_report.csv")
+  end
+
+  def headings
+    ["page", "admin link", "public timestamp", "format", "broken link count", "broken links"]
+  end
+
+  def add_edition_row(csv, edition)
```

Comment:
I think it would be better if this method just returned the csv row for an edition, say, `row_for_edition`, and then the `csv <<` part happens in `generate`.